torch/csrc/Module.cpp

```diff
@@ -524,6 +529,58 @@ PyObject *THPModule_isEnabledXNNPACK(PyObject * /* unused */)
   else Py_RETURN_FALSE;
 }
 
+namespace {
+struct WatchdogThread {
+  WatchdogThread(const std::string& name_, int timeout): name(name_), thrd([this, timeout] {
+      using namespace std::chrono_literals;
+      std::unique_lock<std::mutex> lock(mutex);
+      cv.wait_for(lock, timeout *1s, [this] { return exitFlag; });
+      if (!exitFlag) {
+        std::cerr << "Test " << name << " takes longer than " << timeout << " seconds" <<std::endl;
+        std::exit(-1);
```

Comment:
Are we sure we want a `std::exit` here? Maybe a `std::abort` would be better at actually killing the process?